### PR TITLE
Fix remsql writes which involve multiple index lookups

### PR DESCRIPTION
**Pull request by riverszhang89**
Consider the following queries.

```
CREATE TABLE t (key1 INTEGER, key2 INTEGER)
CREATE INDEX t_key1 ON t(key1)
CREATE INDEX t_key2 ON t(key2)
INSERT INTO t VALUES (val1, val2)
DELETE FROM t WHERE key1 = val1 OR key2 = val2
```

Even though `key1 = val1` and `key2 = val2` find the same row, its genid won't be processed twice for the `skip` shadow table will filter out the duplication. However, there is no such genid deduplication logic in the remsql subsystem. Hence the same delete statement on a foreign table will not work and even crash the source database.

The commit fixes it. The idea is quite simple. A temptable records deleted or updated genid's that have been sent to the foreign database. A remsql operation is skipped if its genid already exists in the temptable.

(DRQS 163943872)

### CODE REVIEW

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d798934/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d798934/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2505'>Resubmit</a></p>

File: db/fdb_fend.c [hunk 1]
```diff
@@ -2633,6 +2643,12 @@ static int fdb_cursor_close(BtCursor *pCur)
               (fdbc->trans) ? fdbc->trans->seq : 0,
               fdbc->fcon.sock.sb);
```
Comment (mistachkin) [inline db/fdb_fend.c:2645]: Does the 'dedup_cur' member need to be closed here via bdb_temp_table_close_cursor() ?

Comment (riverszhang89) [reply db/fdb_fend.c:2645]: `bdb_temp_table_close()` [closes open cursors](https://github.com/bloomberg/comdb2/blob/master/bdb/temptable.c#L1487-L1494). Therefore we need not to explicitly call  `bdb_temp_table_close_cursor()` here.

**Comment by dorinhogea** [comment]
Can we make the dedup temptable and cursor part of  fdb_tran_t?   All transaction stuff goes in there.

File: db/fdb_fend.c (hunk 1 above)
Comment (mistachkin) [reply db/fdb_fend.c:2645]: Ok, in that case, everything looks good.

**Review by mistachkin — APPROVED** [review]
Looks good.

**Comment by riverszhang89** [comment]
@dorinhogea I've moved the dedup temptable to `fdb_tran_t` per your feedback. Can you please take a look again?

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p>The first 10 failing tests are: <br>sc_truncate_lockorder_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/7281034/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/7281034/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2505'>Resubmit</a></p>

**Comment by riverszhang89** [comment]
@mistachkin Thank you for the review!

**Comment by dorinhogea** [comment]
looks good, thank you

**Comment by riverszhang89** [comment]
@dorinhogea Thank you!

**Pull request merged**